cmd/thanos/bucket.go

```diff
@@ -213,4 +217,118 @@ func registerBucket(m map[string]setupFunc, app *kingpin.Application, name strin
 			return printBlock(id)
 		})
 	}
+
+	inspect := cmd.Command("inspect", "inspect all blocks in the bucket")
+	replica := inspect.Flag("replica ", "Filter by replica.").
+		Default("").String()
+	sortBy := inspect.Flag("sort-by", "Sort by columns.").
+		Default("FROM,UNTIL").String()
+
+	m[name+" inspect"] = func(g *run.Group, logger log.Logger, reg *prometheus.Registry, _ opentracing.Tracer, _ bool) error {
+		bucketConfig, err := objStoreConfig.Content()
+		if err != nil {
+			return err
+		}
+
+		bkt, err := client.NewBucket(logger, bucketConfig, reg, name)
+		if err != nil {
+			return err
+		}
+
+		// Dummy actor to immediately kill the group after the run function returns.
+		g.Add(func() error { return nil }, func(error) {})
+
+		defer runutil.CloseWithLogOnErr(logger, bkt, "bucket client")
+
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+		defer cancel()
+
+		// Getting Metas
+		var blockMetas []*block.Meta
+		err = bkt.Iter(ctx, "", func(name string) error {
+			id, ok := block.IsBlockDir(name)
+			if !ok {
+				return nil
+			}
+
+			m, err := block.DownloadMeta(ctx, logger, bkt, id)
+			if err != nil {
+				return err
+			}
+
+			blockMetas = append(blockMetas, &m)
+
+			return nil
+		})
+
+		if err != nil {
+			return err
+		}
+
+		return printTable(blockMetas, replica, sortBy)
+	}
+}
+
+func printTable(blockMetas []*block.Meta, replica *string, sortBy *string) error {
+	header := []string{"ULID", "FROM", "UNTIL", "RANGE", "UNTIL-COMP", "~SIZE", "#SERIES", "#SAMPLES", "#CHUNKS", "COMP-LEVEL", "COMP-FAILED", "REPLICA", "RESOLUTION", "SOURCE"}
+
+	var lines [][]string
+	p := message.NewPrinter(language.English)
+
+	for _, blockMeta := range blockMetas {
+		if *replica != "" && *replica != blockMeta.Thanos.Labels["replica"] {
+			continue
+		}
+
+		var line []string
+		line = append(line, blockMeta.ULID.String())
+		line = append(line, time.Unix(blockMeta.MinTime/1000, 0).Format("02-01-2006 15:04:05"))
+		line = append(line, time.Unix(blockMeta.MaxTime/1000, 0).Format("02-01-2006 15:04:05"))
+		timeRange := time.Duration((blockMeta.MaxTime - blockMeta.MinTime) * 1000 * 1000)
+		line = append(line, timeRange.String())
+		untilComp := "-"
+		if blockMeta.Thanos.Downsample.Resolution == 0 { // data currently raw, downsample if range >= 40 hours
+			untilComp = (time.Duration(40*60*60*1000*time.Millisecond) - timeRange).String()
+		}
+		if blockMeta.Thanos.Downsample.Resolution == 5*60*1000 { // data currently 5m resolution, downsample if range >= 10 days
+			untilComp = (time.Duration(10*24*60*60*1000*time.Millisecond) - timeRange).String()
+		}
+		line = append(line, untilComp)
+		line = append(line, p.Sprintf("%0.2fMiB", (float64(blockMeta.Stats.NumSamples)*1.07)/(1024*1024)))
```

Comment:
My question is: do we really want to hardcode the value `1.07` here? How was this calculated?

Reply:
Afaik it's just an estimation. I saw it on the slides of @bwplotka talk at the OpenSourceSummit. I think the only way to get this really correct is to get the real block size. I just guessed it more or less based on the sample size in my original code.

Reply:
Yea, there are different estimates. Sometimes it is `1.07` sometimes `1.37` sometimes `3` bytes. It all depends what data you hold, how scraping timing is regular etc.

Also this size could be off, because it is only samples size. It does not include the whole index. 
Maybe putting this as constant and stating range of the size `1.07` and `3` would be more accurate if you really want to have a size?

Reply:
I think it's better to remove the size column altogether as it's not easily possible to provide an accurate estimation of the size.